objects/src/builder/mod.rs

```diff
@@ -0,0 +1,274 @@
+use crate::{
+    mock::assembler, Account, AccountCode, AccountError, AccountId, AccountStorage, AccountType,
+    AccountVault, Asset, AssetError, Digest, Felt, FungibleAsset, MerkleStore, NonFungibleAsset,
+    NonFungibleAssetDetails, StorageItem, String, ToString, Vec, ZERO,
+};
+use assembly::ast::ModuleAst;
+use rand::{distributions::Standard, Rng};
+
+pub struct AccountStorageBuider {
+    items: Vec<StorageItem>,
+}
+
+/// Builder for an `AccountStorage`, the builder can be configured and used multipled times.
+impl AccountStorageBuider {
+    pub fn new() -> Self {
+        Self { items: vec![] }
+    }
+
+    pub fn add_item(&mut self, item: StorageItem) -> &mut Self {
+        self.items.push(item);
+        self
+    }
+
+    pub fn build(&self) -> AccountStorage {
+        AccountStorage::new(self.items.clone(), MerkleStore::new()).unwrap()
+    }
+}
+
+/// Builder for an `AccountId`, the builder can be configured and used multipled times.
+pub struct AccountIdBuilder<T: Rng> {
+    account_type: AccountType,
+    on_chain: bool,
+    code_root: Digest,
+    storage_root: Digest,
+    rng: T,
+}
+
+impl<T: Rng> AccountIdBuilder<T> {
+    pub fn new(rng: T) -> Self {
+        Self {
+            account_type: AccountType::RegularAccountUpdatableCode,
+            on_chain: false,
+            code_root: Digest::default(),
+            storage_root: Digest::default(),
+            rng,
+        }
+    }
+
+    pub fn account_type(&mut self, account_type: AccountType) -> &mut Self {
+        self.account_type = account_type;
+        self
+    }
+
+    pub fn on_chain(&mut self, on_chain: bool) -> &mut Self {
+        self.on_chain = on_chain;
+        self
+    }
+
+    pub fn code_root(&mut self, code_root: Digest) -> &mut Self {
+        self.code_root = code_root;
+        self
+    }
+
+    pub fn storage_root(&mut self, storage_root: Digest) -> &mut Self {
+        self.storage_root = storage_root;
+        self
+    }
+
+    pub fn build(&mut self) -> Result<AccountId, AccountError> {
+        let init_seed: [u8; 32] = self.rng.gen();
+        let seed = AccountId::get_account_seed(
+            init_seed,
+            self.account_type,
+            self.on_chain,
+            self.code_root,
+            self.storage_root,
+        )?;
```

Comment:
In some cases (accounts which are not new) we do not care about the account seed and as such we do not need to grind it.  It is sufficient to just randomly search for a valid field element which satisfies the specific accounts requirements (e.g. account type / on(off)-chain  etc).

Reply:
Something like this:

```rust
let regular_id1 = AccountId::new_unchecked((1 << REGULAR_ACCOUNT_SEED_DIGEST_MIN_TRAILING_ZEROS).into());
let regular_id2 = AccountId::new_unchecked((2 << REGULAR_ACCOUNT_SEED_DIGEST_MIN_TRAILING_ZEROS).into());
let regular_id3 = AccountId::new_unchecked((3 << REGULAR_ACCOUNT_SEED_DIGEST_MIN_TRAILING_ZEROS).into());
```

I think we can add that later on if there is the need for it.

Reply:
Yes something like this. The reason I suggest this is because grinding account seeds can be quite a resource intensive process. Defering this for now sounds good, lets keep in mind as we start to integrate and work with the builder.

Reply:
Yeah, fair point! The idea is to do the grinding once and then serialize the data into a file. So the tests would not have to redo any of it. That way we get both consistent account ids and speed. We can still use the accounts with unknown seeds later on though :)

Reply:
Sounds good! :)